fluentui_core/src/main/java/com/microsoft/fluentui/compose/ModalPopup.kt

```diff
@@ -143,14 +186,91 @@ private class ModalWindow(
         setTag(androidx.compose.ui.R.id.compose_view_saveable_id_tag, "Popup:$saveId")
         // Enable children to draw their shadow by not clipping them
         clipChildren = false
+        with(density) { elevation = 8.dp.toPx() }
+        outlineProvider = object : ViewOutlineProvider() {
+            override fun getOutline(view: View, result: Outline) {
+                result.setRect(0, 0, view.width, view.height)
+                result.alpha = 0f
+            }
+        }
     }
 
-    private val windowManager =
-        composeView.context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
+    private var content: @Composable () -> Unit by mutableStateOf({})
+
+    override var shouldCreateCompositionOnAttachedToWindow: Boolean = false
+        private set
+
+    fun show() {
+        windowManager.addView(this, params)
+    }
 
+    fun setCustomContent(
+        parent: CompositionContext? = null,
+        content: @Composable () -> Unit
+    ) {
+        setParentCompositionContext(parent)
+        this.content = content
+        shouldCreateCompositionOnAttachedToWindow = true
+    }
+
+    @Composable
+    override fun Content() {
+        content()
+    }
+
+    private fun focusable(isFocusable: Boolean) = applyNewFlags(
+        if (!isFocusable) {
+            params.flags or WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE
+        } else {
+            params.flags and (WindowManager.LayoutParams.FLAG_NOT_FOCUSABLE.inv())
+        }
+    )
+
+    private fun applyNewFlags(flags: Int) {
+        params.flags = flags
+        popupLayoutHelper.updateViewLayout(windowManager, this, params)
+    }
+
+    fun updateParameters(
+        onDismissRequest: (() -> Unit)?,
+        properties: PopupProperties,
+        layoutDirection: LayoutDirection
+    ) {
+        this.onDismissRequest = onDismissRequest
+        if (properties.usePlatformDefaultWidth && !this.properties.usePlatformDefaultWidth) {
+            params.width = WindowManager.LayoutParams.WRAP_CONTENT
+            params.height = WindowManager.LayoutParams.WRAP_CONTENT
+            popupLayoutHelper.updateViewLayout(windowManager, this, params)
+        }
+        this.properties = properties
+        focusable(properties.focusable)
+        superSetLayoutDirection(layoutDirection)
+    }
+
+    fun updateParentLayoutCoordinates(parentLayoutCoordinates: LayoutCoordinates) {
+        this.parentLayoutCoordinates = parentLayoutCoordinates
+    }
+
+    fun dismiss() {
+        this.setViewTreeLifecycleOwner(null)
+        setViewTreeSavedStateRegistryOwner(null)
+        windowManager.removeViewImmediate(this)
+    }
 
-    private val params: WindowManager.LayoutParams =
-        WindowManager.LayoutParams().apply {
+    override fun setLayoutDirection(layoutDirection: Int) {
```

Comment:
why are we overriding this with No-op

Reply:
For Android versions below API 29, it’s not necessary to explicitly exclude the entire screen from system gestures. However, starting with Android API 29, you need to explicitly specify if you want to exclude the entire screen area from gestures. The skeleton method is defined to keep consistency in the two objects.